src/feltor/feltor.tex

```diff
@@ -2331,7 +2359,56 @@ \subsection{Toroidal ExB angular momentum equation} \label{sec:vorticity_eq}
 \end{longtable}
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\subsection{Parallel momentum balance}
+\subsection{Conservation of Currents (COCE), "COCE": true}\label{sec:COCE}
+If we want to study the electric field locally, without the flux surface average implemented in the equation at section \ref{sec:vorticity_eq}, the COCE diagnostic is necessary. The equation implemented is 
+\begin{equation}
+    \partial_t\Omega-\nc\nc(\boldsymbol{\omega u_E})=\nc(j_\parallel\bhat+\boldsymbol{j_{curv}}+\boldsymbol{j_{b_\perp}})+\Omega_S,
+    \label{eq:COCE_diagnostic}
+\end{equation}   
+with the following definitions:
+\begin{equation}
+\Omega=\Omega_E+\Omega_D=-\nc\boldsymbol{\omega}=-\nc(\boldsymbol{\omega_D}+\boldsymbol{\omega_E})=\nc\left[\frac{m_i}{B^2}\left(\frac{\nabla_\perp p_i}{q_i}+n_i\nabla_\perp\phi\right)\right],
+\end{equation}
+\begin{equation}
+    \Omega_S=\nc\left[\frac{m_i}{B^2}\left(\frac{\nabla_\perp S_{p_i}}{q_i}+S_{n_i}\nabla_\perp\phi\right)\right],
+\end{equation}
+\begin{equation}
+     \boldsymbol{j_{curv}}\approx\frac{\bhat\times\nabla(p_e+p_i)}{B}+(j_\parallel A_{1,\parallel}+\sumsp m_s n_s u_{\parallel,s}^2) \frac{\nabla\times\bhat}{B}=\boldsymbol{j_{d}}+\boldsymbol{j_A}+\boldsymbol{j_\kappa},
+    \label{eq:J_curv_fluid}
+\end{equation}
+\begin{equation}
+      \boldsymbol{j_{b_\perp}}=(j_\parallel+j_{mag})\boldsymbol{b_\perp}-\nc(\boldsymbol{M^{em}}\boldsymbol{b_\perp})=j_\parallel\boldsymbol{b_\perp}+\boldsymbol{j_{M^{em}}}.
+\end{equation}
+
+The implemented terms are defined as follows:
+\begin{longtable}{ll}
+\toprule
+\rowcolor{gray!50}\textbf{Name} &  \textbf{Equation}\\
+\midrule
+    v\_Omega\_E\_gf &$ \mu_i\nc(\frac{N_i\boldsymbol{\nabla_\perp}\phi}{B^2})$ \\
```

Comment:
Also how do you intend to document which divergence is used, I think this must be either updated in code or documented that the toroidal fieldline approx is always used

Reply:
I will recheck the feltordiag entries once the Nablas structure has been tested